core/src/main/java/hudson/model/User.java

```diff
@@ -708,6 +717,32 @@ public static void rekey() {
             or greater issues in the realm change, could affect currently logged
             in users and even the user making the change. */
```

Comment:
Not sure `rekey` has any real test coverage. https://github.com/jenkinsci/jenkins/blob/eab026bdb95ea7006467e150c9cb62ec91ef05d0/test/src/test/java/hudson/model/UserTest.java#L223-L266 do not change the strategy after user records have been created.